anomdetect/src/main/java/com/expedia/adaptivealerting/anomdetect/ShewhartIndividuals.java

```diff
@@ -0,0 +1,185 @@
+    private static final int RECOMPUTE_LIMITS_PERIOD = 25;
+
+    /**
+     * Local Aggregate Moving range. Used to calculate avg. moving range.
+     */
+    private double movingRangeSum;
+
+    /**
+     * Local target predicted from average mean.
+     */
+    private double target;
+
+    /**
+     * Local previous value.
+     */
+    private double prevValue;
+
+    /**
+     * Local total no of received data points.
+     */
+    private int totalDataPoints;
+
+    /**
+     * Local warm up period value. Minimum no of data points required before it can be used for actual anomaly
+     * detection.
+     */
+    private int warmUpPeriod;
+
+    /**
+     * Upper limit for R chart
+     */
+    private double upperControlLimit_R;
+
+    /**
+     * Upper limit for X chart
+     */
+    private double upperControlLimit_X;
+
+    /**
+     * Lower limit for X chart
+     */
+    private double lowerControlLimit_X;
+    /**
+     * Local target
+     */
+    private double runningMean;
+
+    /**
+     * Creates a new MovingRangeChart detector initValue = 0.0, warmUpPeriod = 25
+     */
+    public ShewhartIndividuals() {
+        this(0.0, 25);
+    }
+
+    /**
+     * Creates a new MovingRangeChart detector. Initial target is given by initValue and initial variance is 0.
+     *
+     * @param initValue    Initial observation, used to set the first target estimate.
+     * @param warmUpPeriod Warm up period value. Minimum no of data points required before it can be used for actual anomaly
+     *                     detection.
+     */
+    public ShewhartIndividuals(
+            double initValue,
+            int warmUpPeriod
+    ) {
+        this.prevValue = initValue;
+        this.warmUpPeriod = warmUpPeriod;
+        this.movingRangeSum = 0.0;
+        this.totalDataPoints = 1;
+        this.target = initValue;
+    }
+
+    public int getWarmUpPeriod() {
+        return warmUpPeriod;
+    }
+
+
+    @Override
+    public AnomalyResult classify(MetricPoint metricPoint) {
+        AssertUtil.notNull(metricPoint, "metricPoint can't be null");
+
+        final double observed = metricPoint.value();
+        double currentRange = abs(prevValue - observed);
+        this.movingRangeSum += abs(currentRange);
+        this.runningMean = getRunningMean(observed);
+
+        final double dist = abs(observed - target);
+        this.totalDataPoints++;
+
+        AnomalyLevel anomalyLevel = UNKNOWN;
+        if (totalDataPoints <= warmUpPeriod || ((totalDataPoints-warmUpPeriod) % RECOMPUTE_LIMITS_PERIOD )==0) {
+
+            double averageMovingRange = getAverageMovingRange();
+            this.target = this.runningMean;
+
+            upperControlLimit_R = R_CONTROL_CHART_CONSTANT_D4 * averageMovingRange;
+            upperControlLimit_X = this.target + 2.66 * averageMovingRange;
+            lowerControlLimit_X = this.target - 2.66 * averageMovingRange;
+
+        } else {
+            anomalyLevel = NORMAL;
```

Comment:
We may need to know it's left-tailed and right-tailed (user-defined param) and depending on that we calculate anomaly levels. left-tailed is for lower threshold and right-tailed is for upper threshold.